api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -75,6 +75,15 @@ message HttpConnectionManager {
     PASS_THROUGH = 2;
   }
 
+  enum SchemeHeaderTransformation {
+    // Overwrite any Scheme header with the contents of scheme
+    OVERWRITE_SCHEME = 0;
```

Comment:
The config is still confusing to me.   Can we define it as "oneof"
```
oneof {
// the scheme to override
string  override_scheme = 11;

// preserve downstream one, if absent set it with upstream security level
protobuf.Empty preserve_downstream = 12;

// always set it as upstream security level
protobuf.Empty by_upstream_security_level = 13
}
```

Reply:
I think this is consistent with other header overrides - will let API shephards make that call through :-)